### PR TITLE
Fallback to HPAv1 if HPAv2 not available

### PR DESCRIPTION
**Pull request by LeaveMyYard**
Some older versions of kubernetes does not have HPAv2 API available.

Use HPAv1 API for listing HPA if v2 is not available

### CODE REVIEW

File: robusta_krr/core/integrations/kubernetes.py [hunk 1]
```diff
@@ -216,9 +238,20 @@ async def __list_hpa(self) -> dict[tuple[str, str], HPAData]:
         """
 
         loop = asyncio.get_running_loop()
-        res: V2HorizontalPodAutoscalerList = await loop.run_in_executor(
-            self.executor, lambda: self.autoscaling.list_horizontal_pod_autoscaler_for_all_namespaces(watch=False)
-        )
+
+        try:
+            # Try to use V2 API first
+            res: V2HorizontalPodAutoscalerList = await loop.run_in_executor(
```
Comment (Avi-Robusta) [inline robusta_krr/core/integrations/kubernetes.py:244]: do we need to return the result?

Comment (Avi-Robusta) [reply robusta_krr/core/integrations/kubernetes.py:244]: do we have to format the result to be of object HPAData

**Review by Avi-Robusta — APPROVED** [review]
hey looks good, had a few questions
also wanted to ask if you ran both functions by commenting out some lines just to see they both work?

**Comment by Avi-Robusta** [comment]
also dont forget to rebase

File: robusta_krr/core/integrations/kubernetes.py (hunk 1 above)
Comment (LeaveMyYard) [reply robusta_krr/core/integrations/kubernetes.py:244]: It is formatted below. 
`__get_metric` is the inner function

**Pull request merged**